test/unit-test/mqtt_agent_utest.c

```diff
@@ -258,6 +270,37 @@ MQTTStatus_t MQTT_ProcessLoop_stub( MQTTContext_t * pContext,
     return MQTTSuccess;
 }
 
+/**
+ * @brief A stub for MQTT_ProcessLoop function to be used to test multiple calls to MQTT_ProcessLoop.
+ */
+MQTTStatus_t MQTT_ProcessLoop_stub_multiple( MQTTContext_t * pContext,
+                                             uint32_t timeoutMs,
+                                             int NumCalls )
+{
+    MQTTPacketInfo_t packetInfo;
+    MQTTDeserializedInfo_t deserializedInfo;
+    MQTTAgentContext_t * pMqttAgentContext;
+    MQTTStatus_t status;
+
+    packetInfo.type = packetType;
+    deserializedInfo.packetIdentifier = packetIdentifier;
+
+    pContext->appCallback( pContext, &packetInfo, &deserializedInfo );
```

Comment:
I thought you were only going to call this based on the loop counter?

Reply:
No, its need to be called in both cases for coverage.